src/charm.py

```diff
@@ -231,6 +243,19 @@ def _reconcile_gateway_api_crds(self):
         krm = self._get_resource_manager(GATEWAY_API_CRDS_LABEL)
         krm.reconcile(resources)  # pyright: ignore
 
+    def _publish_oauth_provider_names(self):
+        """Publish the unique OAuth provider name for each ready relation.
+
+        This method iterates over all relations and, if a provider is ready,
+        it publishes its unique OAuth provider name.
+        """
+        for relation in self.ingress_config.relations:
+            if self.ingress_config.is_provider_ready(relation):
+                ingress_app_name = relation.app.name
+                ingress_app_model = self.ingress_config.get_provider_oauth_info(relation).model_name  # type: ignore
+                unique_name = f"oauth-{ingress_app_name}-{ingress_app_model}"
```

Comment:
This is more a purist argument than pragmatist one.  There's no real cost to including model name, but ...

Could we make unique IDs here without model name?  In cases where we really need it I understand why we pass it around, but Juju is designed such that we dont so when possible I'd prefer to avoid it.  I get that model name is included here as a convenient key to make this unique, but what if we do something like:

```
unique_name = f"oauth-{ingress_app_name}-{hash(ingress_config.settings)}"
```

That'll be unique, deterministic, and if there's ever a collision then they're true duplicates anyway